Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -242,33 +242,33 @@ Value Value::to_number(Interpreter& interpreter) const
     ASSERT_NOT_REACHED();
 }
 
-double Value::to_double(Interpreter& interpreter) const
+i32 Value::as_i32() const
 {
-    auto number = to_number(interpreter);
-    if (interpreter.exception())
-        return 0;
-    return number.as_double();
+    return static_cast<i32>(as_double());
 }
 
-i32 Value::to_i32() const
+size_t Value::as_size_t() const
```

Comment:
Maybe this should assert that the value is non-negative instead of giving you 0?

Reply:
Yep, that seems reasonable